### PR TITLE
feat: show failed payment banner and block the workspace integration 

### PR DESCRIPTION
**Pull request by vikrantgupta25**
### Summary

- added a new endpoint for query service to serve the active license rather than frontend writing searches on arrays to find the same -> `v3/licenses/active` 
- added `AppContext` to only fetch the required details once rather than fetching them multiple times across multiple components. the `AppContext` will be later used to migrate everything from store. 
- added a new route `workspace-suspended` to handle all the suspensions based on events. the current route is not made scalable. 
- when the event queue contains the `FAILED_PAYMENT` event then show the warning banner.
- when the status of license is `SUSPENDED` and state is `PAYMENT_FAILED` then suspend the workspace

#### Related Issues / PR's

image

image

#### Screenshots

NA

#### Affected Areas and Manually Tested Areas

----

> [!IMPORTANT]
> Introduces a new feature to handle failed payments by adding an endpoint, context, and UI components, including a new route for workspace suspension.
> 
>   - **Behavior**:
>     - New endpoint `v3/licenses/active` in `api.go` to fetch active license.
>     - Added `AppContext` to fetch required details once, reducing multiple fetches.
>     - New route `workspace-suspended` for handling workspace suspensions.
>   - **Frontend**:
>     - Added `WorkspaceSuspended` component in `WorkspaceSuspended.tsx` with styles in `WorkspaceSuspended.styles.scss`.
>     - Updated `AppLayout` to show payment failed banner and handle billing.
>     - Added translations for failed payment messages in `failedPayment.json`.
>   - **API**:
>     - Added `getActiveLicenseV3` function in `license.go` to fetch active license data.
>     - Created `getActive` function in `getActive.ts` to call the new API endpoint.
>   - **Routes and Context**:
>     - Updated `routes.ts` and `pageComponents.ts` to include `WorkspaceSuspended`.
>     - Added `useActiveLicenseV3` hook and `AppProvider` for context management.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=SigNoz%2Fsignoz&utm_source=github&utm_medium=referral)<sup> for 016f6bafd4ae7845054da558d373e4158fabdf34. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by vikrantgupta25** [comment]
contributes to - https://github.com/SigNoz/platform-pod/issues/306

**Comment by YounixM** [comment]
1. Create a ticket to show timezone in workspace locked message.
2. Create a ticket to show timezone in workspace suspended message.

File: frontend/src/AppRoutes/routes.ts [hunk 1]
```diff
@@ -364,6 +365,13 @@ const routes: AppRoutes[] = [
 		isPrivate: true,
 		key: 'WORKSPACE_LOCKED',
 	},
+	{
+		path: ROUTES.WORKSPACE_SUSPENDED,
+		exact: true,
+		component: WorkspaceSuspended,
+		isPrivate: true,
+		key: 'WORKSPACE_LOCKED',
```
Comment (YounixM) [inline frontend/src/AppRoutes/routes.ts:373]: Do we want to have the same key for this route too ?

File: frontend/src/api/licensesV3/getActive.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { ApiV3Instance as axios } from 'api';
+import { ErrorResponseHandler } from 'api/ErrorResponseHandler';
+import { AxiosError } from 'axios';
+import { ErrorResponse, SuccessResponse } from 'types/api';
+import { LicenseV3EventQueueResModel } from 'types/api/licensesV3/getActive';
+
+const getActive = async (): Promise<
+	SuccessResponse<LicenseV3EventQueueResModel> | ErrorResponse
+> => {
+	try {
```
Comment (YounixM) [inline frontend/src/api/licensesV3/getActive.ts:10]: Let's remove the try catch block and handle the error in the consuming flow.

File: frontend/src/container/AppLayout/index.tsx [hunk 3]
```diff
@@ -289,6 +349,36 @@ function AppLayout(props: AppLayoutProps): JSX.Element {
 					)}
 				</div>
 			)}
+			{!showTrialExpiryBanner && showPaymentFailedWarning && (
+				<div className="payment-failed-banner">
+					Your bill payment has failed. Your workspace will get suspended on{' '}
+					<span>
```
Comment (YounixM) [inline frontend/src/container/AppLayout/index.tsx:355]: Check light mode

Comment (vikrantgupta25) [reply frontend/src/container/AppLayout/index.tsx:359]: donee

File: frontend/src/api/licensesV3/getActive.ts (hunk 2 above)
Comment (vikrantgupta25) [reply frontend/src/api/licensesV3/getActive.ts]: done

**Review by YounixM — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Pull request merged**